assertk/src/jvmMain/kotlin/assertk/failure.kt

```diff
@@ -12,9 +12,5 @@ internal actual inline fun failWithNotInStacktrace(error: Throwable): Nothing {
     throw error
 }
 
-@Suppress("EXTENSION_SHADOWED_BY_MEMBER")
-internal actual inline fun Throwable.addSuppressed(error: Throwable) {
-    throw NotImplementedError()
-}
-
+@PublishedApi
 internal actual inline fun Throwable.isOutOfMemory(): Boolean = this is OutOfMemoryError
```

Comment:
You might want to consider renaming this `isFatal` or something and using `VirtualMachineError` as a check which also would include things like `StackOverflowError` and `LinkageError` that are basically unrecoverable.

Retrofit gets its set of fatals from RxJava which is: https://github.com/square/retrofit/blob/c187f4a1d6a63b302040df4b7736e6ce024ac100/retrofit/src/main/java/retrofit2/Utils.java#L529-L539